src/pages/[guild]/members.tsx

```diff
@@ -141,17 +141,21 @@ const GuildPage = (): JSX.Element => {
   )
 
   useEffect(() => {
-    /**
-     * Using native browser api, so it doesn't cause a rerender & trigger the general
-     * loading bar set up in _app.tsx. The downside is, if we filter for something,
-     * navigate to another page then click the back button, it won't work. We could
-     * add some logic to the _app.tsx effect to filter out query-only changes
-     * (https://stackoverflow.com/questions/62368109/update-router-query-without-firing-page-change-event-in-next-js),
-     * but I'm not sure the performance cost would worth handling this edge case, so
-     * left it this way for now
-     */
-    window.history.pushState("", "", `?${queryString}`)
-  }, [queryString])
+    if (!router.isReady) return
+
+    const searchParams = new URLSearchParams(queryString)
+    const query = {}
+    searchParams.forEach((value, key) => {
+      query[key] = query[key] || []
+      query[key].push(value)
+    })
+
+    router.replace({
+      pathname: router.pathname,
```

Comment:
Do we need this line? IIRC we can just pass replace what we actually wan't to replace, no?

Reply:
We could instead of all this, simply just do
```
  useEffect(() => {
    if (!router.isReady || !queryString) return

    const asPath = router.asPath.split("?")[0]
    router.replace(`${asPath}?${queryString}`)

  }, [queryString, router.isReady])
```

I did not want to depend on string splitting, and then it just escalated.